File: src/main/java/lesson3/Cat.java
package lesson3;

class Cat implements Participiant {
    static int maxJumpHeight = 20;
    static int maxRunDistance = 30;

    @Override
    public boolean jump(int height) {
        if (height <= Cat.maxJumpHeight) {
            System.out.println("Кот успешно прыгнул");
            return true;
        }
        System.out.println("Кот не прыгнул");
        return false;
    }

    @Override
    public boolean run(int distance) {
        if (distance <= Cat.maxRunDistance) {
            System.out.println("Кот успешно пробежал");
            return true;
        }
        System.out.println("Кот не пробежал");
        return false;
    }
}

File: src/main/kotlin/lesson3_kt/Cat.kt
package lesson3_kt

class Cat : Participian{
    companion object {
        val CAT_MAX_DISTANCE = 20
        val CAT_MAX_HEIGHT = 30
    }
    override fun run(distance: Int) : Boolean = performAction(
        value = distance,
        maxValue = CAT_MAX_DISTANCE,
        action = "пробежал",
        actor = "кот"
    )

    override fun jump(height: Int) : Boolean= performAction(
        value = height,
        maxValue= CAT_MAX_HEIGHT,
        action = "прыгнул",
        actor = "кот"
    )
}